packages/auth/amplify_auth_cognito/lib/src/state/cognito_state_machine.dart

```diff
@@ -25,6 +25,7 @@ import 'state.dart';
 final stateMachineBuilders = <StateMachineToken, StateMachineBuilder>{
   AuthStateMachine.type: AuthStateMachine.new,
   CredentialStoreStateMachine.type: CredentialStoreStateMachine.new,
+  FetchAuthSessionStateMachine.type: FetchAuthSessionStateMachine.new,
```

Comment:
Minor nit, perhaps it's too late, but naming seems a bit misleading. Shouldn't it be cognito_state_machine_manager to denote it's a state machine manager instead of a single state machine instance? I had to read through the code and see that it extends StateMachineManager before being certain.

Reply:
To an external observer, it is a single state machine. The goal is to abstract away the fact that it manages multiple state machines because all communication should happen with this state machine.